src/UI/Implementation/Component/Button/Renderer.php

```diff
@@ -54,14 +54,33 @@ protected function render_button(Component\Component $component, RendererInterfa
 			$tpl->setVariable("DISABLED", self::DISABLED_CLASS);
 		}
 
+		$this->maybe_render_id($component, $tpl);
+
 		return $tpl->get();
 	}
 
-	protected function render_close() {
+	protected function render_close($component) {
 		$tpl = $this->getTemplate("tpl.close.html", false, false);
+		$this->maybe_render_id($component, $tpl);
 		return $tpl->get();
 	}
 
+	protected function maybe_render_id($component, $tpl) {
+		$id = $this->bindJavaScript($component);
+		if ($id !== null) {
+			$tpl->setVariable("ID", "id=\"$id\"");
+		}
+		else {
+			// This is required for the close button. I tried to use the purge_unfilled_vars
+			// option for getTemplate in render_close, but it doesn't do what i expect. I
+			// checked the passing of the parameter down to HTML_Template_IT, which also
+			// works. There seems to be some arcane spell or dance, that HTML_Template_IT
+			// want me to perform, that i don't know.
+			// -- klees, 2016-08-17
+			$tpl->setVariable("ID", "");
```

Comment:
I do not think you understood me correctly. 

To me it seemed to be more transparent to leave the templating stuff to the implementor of the Renderer completely, thus stop at the generation of the id.

Fully agreed. But there are Components that my have ID's (such as Default Buttons). Those Components will provide logic to set the ID in the templates (like default Butons do). Why not just give therefore all default Buttons an ID?

I do not have strong opinion on this and as pointed out further above, both ways have pros and cons.